.github/workflows/build_devcontainer.yml

```diff
@@ -0,0 +1,29 @@
+name: "build-devcontainer"
+on: # rebuild any PRs and main branch changes
+  pull_request:
```

Comment:
Why build on PR ? It should already have been built in the latest push before the PR

Reply:
Building on PR changes to .devcontainer is fine, but then shouldn't you remove the `push:` statement ?